iio_widget.c

```diff
@@ -77,6 +77,8 @@ static void iio_widget_init(struct iio_widget *widget,
 	widget->update_value = update_value;
 	widget->save = save;
 	widget->priv = priv;
+	widget->priv_progress = NULL;
+	widget->priv_convert_function = NULL;
```

Comment:
would a `memset(widget, 0, sizeof(*widget))` work instead at the beginning of this init block ?
the added benefit is that if you extend the `iio_widget` struct later, you can be sure that the new fields are 0/NULL if un-initialized explicitly;

Reply:
Yes, that will be better.